### DISCUSSION TITLE
Animating Particle Trajectory File in Diffusion Tutorial

### DISCUSSION THREAD

**Discussion by ttoshiro** [Parcels-code/Parcels#1408] [Q&A] [answered]
I would like to animate the particle trajectories outlined in the Milton scheme of order 1 (M1) advection-diffusion simulation that is "walked through" (haha) in the Diffusion Tutorial on the Ocean Parcels website. While I am able to a static trajectory plot, shown below, attempting to animate this results in a TypeError, also displayed.

Link to the Diffusion Tutorial: https://docs.oceanparcels.org/en/latest/examples/tutorial_diffusion.html

Static particle trajectory plot:
![download-5](https://github.com/OceanParcels/parcels/assets/66069264/39875a81-abf8-4445-979e-c10c6a169956)

Code attempting to animate this:
`plotTrajectoriesFile("M1_high.zarr", mode="movie2d")`

Error:
<img width="1127" alt="Screen Shot 2023-08-03 at 5 34 40 PM" src="https://github.com/OceanParcels/parcels/assets/66069264/ff62ff91-1a8a-475d-b75b-dfe7b25852c0">

Note that, in the code above, M1_high.zarr is equivalent to the M1_out.zarr file in the diffusion tutorial. Any ideas on what to do?

**Comment by erikvansebille** [comment, marked as answer]
Thanks for reporting, @ttoshiro. The built-in plotting functionality in Parcels was always very rudimentary, and not very robust. We've therefore removed it recently in #1397. 

I suggest that you write your own code to make an animation, for example based on the code at https://docs.oceanparcels.org/en/latest/examples/tutorial_output.html#Animations
